### ISSUE TITLE
Colour highlighting/folding for inheritance items in Inspector

### ISSUE THREAD

**Issue by chucklepie** [godotengine/godot-proposals#3196]
### Describe the project you are working on

2D

### Describe the problem or limitation you are having in your project

The resource box colour in the Inspector makes life much simpler. Could this be extended to support nested inherited items?

The following only has 1 level of inheritance but it's easy to not notice they come from different classes. If there was more than this level of inheritance it would be even harder.
![image](https://user-images.githubusercontent.com/12863685/130680136-8df3208d-ee48-42dc-949d-8b75b71051cb.png)

e.g. as below, but what would be really nicer would also be a means of folding them...

![image](https://user-images.githubusercontent.com/12863685/130680399-6a445555-39a6-4163-87c0-a65a22a0d391.png)

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

It's sometimes important to know which class data belongs too and sometimes it's nice to be able to not see some of levels

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

As above images really, with an added bonus accumulator for folding.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

It can't

### Is there a reason why this should be core and not an add-on in the asset library?

It's an IDE change

**Comment by KoBeWi** [comment]
This was already implemented in 4.0: https://github.com/godotengine/godot/pull/32428

**Comment by pycbouh** [comment]
@KoBeWi You sure? This proposal seems to be about the remote inspector, and the PR is not from the looks of it.

**Comment by KoBeWi** [comment]
It never mentions remote though.

**Comment by pycbouh** [comment]
True, but it's the only way I know to achieve the pictured result.

**Comment by KoBeWi** [comment]
@chucklepie I'd suggest to check if this proposal is satisfied in 4.0 (you can get a semi-official build [here](https://downloads.tuxfamily.org/godotengine/testing/4.0/)). If not, it can be reopened.

**Comment by pycbouh** [comment]
It's still an issue in master, so reopening:

Local is fine:
![godot windows tools 64_2021-08-25_18-15-25](https://user-images.githubusercontent.com/11782833/130817617-9aacc4d4-fd09-4f3a-92f3-9e78e40a9eec.png)

But not remote:
![godot windows tools 64_2021-08-25_18-15-40](https://user-images.githubusercontent.com/11782833/130817624-d66b05c5-3b6e-48c2-989b-3409df8ad81d.png)
